### PR TITLE
gh-95005: Replace PyAccu with PyUnicodeWriter

### PR DESCRIPTION
**Pull request by aivarsk**
PyAccu was still being used in only two places: the JSON encoder and StringIO.
Replace PyAccu with PyUnicodeWriter and remove PyAccu completely.

* Issue: gh-95005

### CODE REVIEW

**Review by MaxwellDupre — APPROVED** [review]
only 3 tests failed from test suite. Code is good.

**Comment by corona10** [comment]
@MaxwellDupre

> only 3 tests failed from test suite. Code is good.

Which test?

**Comment by aivarsk** [comment]
buildbot/AMD64 Debian seems to have a broken compiler or hardware error. The compiler crashed the last time and now it gets SIGBUS while compiling `classobject.c` which was not modified by the patch.

**Comment by MaxwellDupre** [comment]
I ran 
./python -m test -j0 -W -uall
because of the code coverage and out of >400 tests 3 failed which IMHO was insignificant also those fails did not seem to be related to the fix. Hence, I didn't record them and thought not worth mentioning.

**Comment by corona10** [comment]
> buildbot/AMD64 Debian seems to have a broken compiler or hardware error. The compiler crashed the last time and now it gets SIGBUS while compiling classobject.c which was not modified by the patch.

I sent a mail to the machine owner, seems unrelated to this PR.

File: Modules/_json.c [hunk 1]
```diff
@@ -1280,19 +1279,29 @@ encoder_call(PyEncoderObject *self, PyObject *args, PyObject *kwds)
...
+    if (encoder_listencode_obj(self, &writer, obj, indent_level)) {
+        _PyUnicodeWriter_Dealloc(&writer);
         return NULL;
-    if (encoder_listencode_obj(self, &acc, obj, indent_level)) {
-        _PyAccu_Destroy(&acc);
+    }
+
+    result = PyTuple_New(1);
+    if (result == NULL ||
+            PyTuple_SetItem(result, 0, _PyUnicodeWriter_Finish(&writer))) {
```
Comment (corona10) [inline Modules/_json.c]:
```suggestion
            PyTuple_SetItem(result, 0, _PyUnicodeWriter_Finish(&writer) < 0)) {
```

**Review by corona10 — APPROVED** [review]
lgtm 

Overall looks good, I just left nit comment.

**Pull request merged**